nebula_ros/src/continental/continental_ars548_decoder_ros_wrapper.cpp

```diff
@@ -0,0 +1,851 @@
+    point.classification_bicycle = object.classification_bicycle;
+    point.classification_pedestrian = object.classification_pedestrian;
+    point.dynamics_abs_vel_x = static_cast<float>(object.absolute_velocity.x);
+    point.dynamics_abs_vel_y = static_cast<float>(object.absolute_velocity.y);
+    point.dynamics_rel_vel_x = static_cast<float>(object.relative_velocity.x);
+    point.dynamics_rel_vel_y = static_cast<float>(object.relative_velocity.y);
+    point.shape_length_edge_mean = object.shape_length_edge_mean;
+    point.shape_width_edge_mean = object.shape_width_edge_mean;
+    point.dynamics_orientation_rate_mean = object.orientation_rate_mean;
+
+    output_pointcloud->points.emplace_back(point);
+  }
+
+  output_pointcloud->height = 1;
+  output_pointcloud->width = output_pointcloud->points.size();
+  return output_pointcloud;
+}
+
+radar_msgs::msg::RadarScan ContinentalARS548DriverRosWrapper::ConvertToRadarScan(
+  const continental_msgs::msg::ContinentalArs548DetectionList & msg)
+{
+  radar_msgs::msg::RadarScan output_msg;
+  output_msg.header = msg.header;
+  output_msg.returns.reserve(msg.detections.size());
+
+  radar_msgs::msg::RadarReturn return_msg;
+  for (const auto & detection : msg.detections) {
+    if (
+      detection.invalid_azimuth || detection.invalid_distance || detection.invalid_elevation ||
+      detection.invalid_range_rate) {
+      continue;
+    }
+
+    return_msg.range = detection.range;
+    return_msg.azimuth = detection.azimuth_angle;
+    return_msg.elevation = detection.elevation_angle;
+    return_msg.doppler_velocity = detection.range_rate;
+    return_msg.amplitude = detection.rcs;
+    output_msg.returns.emplace_back(return_msg);
+  }
+
+  return output_msg;
+}
+
+radar_msgs::msg::RadarTracks ContinentalARS548DriverRosWrapper::ConvertToRadarTracks(
+  const continental_msgs::msg::ContinentalArs548ObjectList & msg)
+{
+  radar_msgs::msg::RadarTracks output_msg;
+  output_msg.tracks.reserve(msg.objects.size());
+  output_msg.header = msg.header;
+
+  constexpr int16_t UNKNOWN_ID = 32000;
+  constexpr int16_t CAR_ID = 32001;
+  constexpr int16_t TRUCK_ID = 32002;
+  constexpr int16_t MOTORCYCLE_ID = 32005;
+  constexpr int16_t BICYCLE_ID = 32006;
+  constexpr int16_t PEDESTRIAN_ID = 32007;
+  constexpr float INVALID_COVARIANCE = 1e6;
+
+  radar_msgs::msg::RadarTrack track_msg;
+  for (const auto & object : msg.objects) {
+    track_msg.uuid.uuid[0] = static_cast<uint8_t>(object.object_id & 0xff);
+    track_msg.uuid.uuid[1] = static_cast<uint8_t>((object.object_id >> 8) & 0xff);
+    track_msg.uuid.uuid[2] = static_cast<uint8_t>((object.object_id >> 16) & 0xff);
+    track_msg.uuid.uuid[3] = static_cast<uint8_t>((object.object_id >> 24) & 0xff);
+
+    const double half_length = 0.5 * object.shape_length_edge_mean;
+    const double half_width = 0.5 * object.shape_width_edge_mean;
+    const int reference_index = std::min<int>(object.position_reference, 8);
```

Comment:
A comment like "position references 0-7 are valid. In case of invalid references, use the dummy reference at index 8" would make this more clear imo!

Reply:
Added documentation in [ec80eab](https://github.com/tier4/nebula/pull/123/commits/ec80eab73f575aad21c9521e0db734a0474ef639)